README.adoc

```diff
@@ -471,6 +471,79 @@ To remove the image, run the following command:
 docker rmi openliberty-getting-started:1.0-SNAPSHOT
 ```
 
+// =================================================================================================
+// Developing the application in a Docker container
+// =================================================================================================
+
+== Developing the application in a Docker container
+
+// File 0
+server.xml
+[source, xml, linenums, role='code_column']
+----
+include::finish/src/main/liberty/config/server.xml[]
+----
+
+The Open Liberty Maven plug-in includes a `devc` goal that simplifies developing
+your application in a Docker container. This goal builds a Docker
+image, mounts the required directories, binds the required ports, and then runs
+the application inside of a container. This development mode, known as dev mode, also listens for
+any changes in the application source code or configuration and rebuilds the
+image and restarts the container as necessary.
+
+Build and run the container by running the devc goal from the `start` directory:
+
+[role='command']
+```
+mvn liberty:devc
+```
+
+After you see the following message, your application server in dev mode is ready:
+
+[source, role="no_copy"]
+----
+************************************************************************
+*    Liberty is running in dev mode.
+----
+
+Open another command-line session and run the `docker ps` command to verify that your container started:
+[role='command']
+```
+docker ps
+```
+
+Make sure that your container is running and does not have `Exited` as its status:
+
+[role="no_copy"]
+----
+CONTAINER ID        IMAGE                                 COMMAND                  CREATED             STATUS                         PORTS                                                                    NAMES
+17af26af0539        guide-getting-started-dev-mode        "/opt/ol/helpers/run…"   3 minutes ago       Up 3 minutes                   0.0.0.0:7777->7777/tcp, 0.0.0.0:9080->9080/tcp, 0.0.0.0:9443->9443/tcp   liberty-dev
+----
+
+To access the application, go to the http://localhost:9080/system/properties[http://localhost:9080/system/properties^] URL. 
+
+Similar to the development mode used previously in this guide, this
+development mode automatically picks up changes that you make to your
+application and allows you to run tests by pressing the `enter/return` key in the
+active command-line session.
+
+Update the [hotspot file=0]`server.xml` file to change the context root from `/` to `/dev`.
+
+[role="code_command hotspot", subs="quotes"]
+----
+#Replace the server configuration file.#
+`src/main/liberty/config/server.xml`
+----
+
+After you make the file changes, Open Liberty automatically reloads its
+configuration. You will now be able to access the application at the
```

Comment:
You will now be able to access the application...
-->
You can access the application...